packages/flutter/lib/src/material/carousel.dart

```diff
@@ -190,9 +193,12 @@ class CarouselView extends StatefulWidget {
     this.reverse = false,
     this.consumeMaxWeight = true,
     this.onTap,
+    this.disabledChildrenInteraction = true,
     required List<int> this.flexWeights,
     required this.children,
-  }) : itemExtent = null;
+  })  : itemExtent = null,
+        assert(disabledChildrenInteraction == true ||
+            (disabledChildrenInteraction == false && onTap == null));
```

Comment:
Could be simplified to:

```suggestion
        assert(disabledChildrenInteraction || onTap == null));
```

Or alternatively, instead of `assert`ing, we could just wrap in a `GestureDetector` when there's a tap callback without a splash.

Reply:
Thank you so much for the suggestion !